spec/models/host_factory_spec.rb

```diff
@@ -88,5 +88,58 @@
         expect(host.role.memberships_as_member.map(&:role)).to eq(host_factory.role.layers)
       end
     end
+
+    describe 'verify create host given AUTHN_API_KEY config' do
+      context 'when CONJUR_AUTHN_API_KEY_DEFAULT is true' do
+        before do
+          allow(ENV).to receive(:[]).with('CONJUR_AUTHN_API_KEY_DEFAULT').and_return('true')
+          Rails.application.config.conjur_config.authn_api_key_default = true
+        end
+
+        context 'when creating host with api-key annotation true' do
+          let(:options) { {annotations: {'authn/api-key' => true}} }
+          it { expect { host_builder.create_host }.to_not raise_error }
+        end
+
+        context 'when creating host with api-key annotation false' do
+          let(:options) { {annotations: {'authn/api-key' => false}} }
+          it { expect { host_builder.create_host }.to_not raise_error }
+        end
+
+        context 'when creating host without api-key annotation' do
+          it { expect { host_builder.create_host }.to_not raise_error }
+        end
+      end
+
+      context 'when CONJUR_AUTHN_API_KEY_DEFAULT is false' do
+        before do
+          allow(ENV).to receive(:[]).with('CONJUR_AUTHN_API_KEY_DEFAULT').and_return('false')
+          Rails.application.config.conjur_config.authn_api_key_default = false
+        end
+
+        after do
+          Rails.application.config.conjur_config.authn_api_key_default = true
```

Comment:
You can avoid the after block in the 2 spec files in this PR by stubbing the `authn_api_key_default` in the before block.
```ruby
allow(Rails.application.config.conjur_config).to receive(:authn_api_key_default).and_return('false')
```

Reply:
The good thing about stubbing is that it does the clean up for you

Reply:
@doodlesbykumbi , It is currently stubbed in the before block, no? 
before do
     allow(ENV).to receive(:[]).with('CONJUR_AUTHN_API_KEY_DEFAULT').and_return('false')
     Rails.application.config.conjur_config.authn_api_key_default = false
 end
When I don't put the cleanup in the after block, other tests in host_factory_spec.rb occasionally failed with message related to the config set to false. Maybe the cleanup happens only at the end of the entire class?

Reply:
@egvili I think ` Rails.application.config.conjur_config.authn_api_key_default = false` doesn't count as stubbing, since it explicitly setting the value. I might be missing something but I thought instead using `allow(..).and_return(...)` results in rspec intercepting the calls only for the duration of the test.

Reply:
@doodlesbykumbi , I now understand what you meant. Please check now